x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/install_prebuilt_rules_and_timelines/install_prebuilt_rules_package.ts

```diff
@@ -19,24 +22,45 @@ export async function installPrebuiltRulesPackage(
   config: ConfigType,
   context: SecuritySolutionApiRequestHandlerContext
 ) {
-  // Get package version from the config
+  const pkgVersion = await findLatestPackageVersion(config, context, PREBUILT_RULES_PACKAGE_NAME);
+
+  return context
+    .getInternalFleetServices()
+    .packages.ensureInstalledPackage({ pkgName: PREBUILT_RULES_PACKAGE_NAME, pkgVersion });
+}
+
+export async function installEndpointPackage(
+  config: ConfigType,
+  context: SecuritySolutionApiRequestHandlerContext
+) {
+  const pkgVersion = await findLatestPackageVersion(config, context, ENDPOINT_PACKAGE_NAME);
+
+  return context.getInternalFleetServices().packages.ensureInstalledPackage({
+    pkgName: ENDPOINT_PACKAGE_NAME,
+    pkgVersion,
+  });
+}
+
+async function findLatestPackageVersion(
+  config: ConfigType,
+  context: SecuritySolutionApiRequestHandlerContext,
+  packageName: string
+) {
   let pkgVersion = config.prebuiltRulesPackageVersion;
 
   // Find latest package if the version isn't specified in the config
   if (!pkgVersion) {
+    const securityAppClient = context.getAppClient();
     // Use prerelease versions in dev environment
     const isPrerelease =
-      context.getAppClient().getKibanaVersion().includes('-SNAPSHOT') ||
-      context.getAppClient().getKibanaBranch() === 'main';
+      securityAppClient.getBuildFlavor() === 'traditional' &&
```

Comment:
Why is this additional check needed? Would it otherwise be possible to get a prerelease version in serverless, somehow?

Reply:
Prerelease packages are considered unstable, so should not be installed in Serverless. There's some context in this PR: https://github.com/elastic/kibana/pull/170975#issue-1986052975

It should still be possible to install a prerelease version of a package by calling the Fleet API directly, if needed.